src/base/Solution.cpp

```diff
@@ -51,6 +51,18 @@ void Solution::setTransport(shared_ptr<Transport> transport) {
     m_transport = transport;
 }
 
+void Solution::setTransport(const std::string& model) {
+    if (model == "") {
+        setTransport(shared_ptr<Transport>(
+            newDefaultTransportMgr(m_thermo.get())));
+    } else if (model == "None") {
+        setTransport(shared_ptr<Transport>(newTransportMgr("None")));
```

Comment:
Is this special case needed?

Reply:
I included this as it was already included in the passage I moved over from `newSolution` (which now calls this new method). If I recall, the original point was to not have dangling pointers within the `Solution` object, so all exceptions can be thrown as needed.

Reply:
Alright. The short answer is: no, special cases is not needed :tada: